### ISSUE TITLE
Open file opens in a new vscode instance

### ISSUE THREAD

**Issue by thecodingdude** [Microsoft/vscode#9479]
- VSCode Version: 1.3.1
- OS Version: Windows 8.1

Steps to Reproduce:

1. Open one VS code instance with some tabs
2. Open file from OS (windows explorer)

Expected Result: File from explorer opens in last active window

Actual Result: New VSCode instance opens

Related issues: [#3191](https://github.com/Microsoft/vscode/issues/3191)

Comments: Agreed with @thewhitetulip. This breaks the expected UX flow, no other editor/IDE (I.E sublime) opens a new instance if one is already open. Using CLI arguments is an option, but opening a new instance should be a flag, not opening in the last active window.

**Comment by thewhitetulip** [comment]
@egamma

**Comment by egamma** [comment]
This flag already exists:

```json
     "window.openFilesInNewWindow": false
```

**Comment by thewhitetulip** [comment]
@egamma The question here is, why isn't this the default behavior?

**Comment by egamma** [comment]
@thewhitetulip clarifying the issue title. 

I don't have the full history on this setting, but I remember that there where some strong opinions that this setting should be true by default.

Assigning to @bpasero

**Comment by thewhitetulip** [comment]
@egamma Yes, it'll be a major time saver if we have this as default! I have to use TextWrangler just because I can't open things inside vscode from outside (the only way is to drag and drop files inside it)

**Comment by egamma** [comment]
@thewhitetulip not sure I understand the time saver argument. You change the user settings once and you have the behaviour you want.

**Comment by thewhitetulip** [comment]
1. There are probably millions of people using VSCode.
2. When everyone tries to right click and open a file in vscode, it opens in a new instance
3. They go to the Internet and change the setting manually, takes them few minutes to understand and make the change

What can be fixed in one minute, will take one million minutes for everyone to fix.
Isn't this a time saver?

Also isn't this the default behavior of literally every other text editor in existence?

**Comment by egamma** [comment]
From the git history I looks like the setting is as is since at least last November and people haven't complained, they might complain if the default is getting changed. 

Need to waiting for the insights of @bpasero with regard to the default value of this setting.

**Comment by thewhitetulip** [comment]
@egamma I am sorry, there are two issues opened on this issue tracker related to the same thing. I do not think we can conclude that _people haven't complained_. :smile:

**Comment by egamma** [comment]
@thewhitetulip 😄, but my point was that changing the default value of a setting should not be done lightly, it needs to consider the history, impact on users that did not complain etc.

**Comment by thewhitetulip** [comment]
Well, to be honest, for any editor, if you open a new file from the file explorer, the default behaviour has to be to open in the existing window. I can't think of a single editor who doesn't do this!

Also I can't think of a single person who won't expect this, so the users who didn't complain may have never have to use this feature, I am sure they'll hate that this feature doesn't exist!

**Comment by bpasero** [comment]
I agree with @egamma on making this decision with care and I also think we should not just change it without looking at how other editors behave. We can also not just look at Windows alone because on Mac you might have a well different behaviour. Also, this setting has a greater impact than just opening from the windows explorer, it has an impact on every place where files get opened including the command line. We might not even be able to distinguish how a file got opened (from the context menu of the windows explorer, via double click, via command line, etc).

**Comment by thecodingdude** [comment]
@bpasero I appreciate what you're saying but the reality is the current implementation simply breaks UI flow. Chrome, Firefox, Internet Explorer, Edge, Safari and more all have tabs and do not open a new instance every time a page is opened, from explorer or otherwise.

It is expected that anything 'tabbed' opens in the current tabset rather than spawning a new instance of the program. Imagine if browsers did that....Now think about VS Code....

**Comment by bpasero** [comment]
There seems to be platform differences, e.g. on Mac I see a new window opening for each file I click:

![editors](https://cloud.githubusercontent.com/assets/900690/17612993/857507ee-6058-11e6-92d6-1f8a0c421958.gif)

On Windows, I see different behaviour. 

If we revisit this, it has to be OS specific and it should cover all combinations of how a file can be opened (as outlined in my comment in https://github.com/Microsoft/vscode/issues/10390#issuecomment-238889820)

**Comment by olaeld** [comment]
There is also an issue that, after VS Code is already open, right-clicking on a file in Windows and selecting "Open with Code", VS Code does not appear in front, so I need to perform a second action to switch. Clearly, this is not what I want.
